NEWS.md

```diff
@@ -26,6 +26,8 @@
 
 * `redundant_equals_linter()` for redundant comparisons to `TRUE` or `FALSE` like `is_treatment == TRUE` (#1500, @MichaelChirico)
 
+* `return_assignment_linter()` for assignments within the `return()` clause of a function, e.g. `return(x <- foo())` (@MichaelChirico)
+
```

Comment:
Nice!

Do you think implicit return assignments should also lint? I have seen code like this in the wild:

``` r
foo <- function() {
  x <- 4
}
y <- foo() 
y
#> [1] 4
```

<sup>Created on 2022-09-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Not meant to be resolved in this PR, but looking at it reminded me of this.

Reply:
Good point that outside Google this form will be a lot more common (internally, we require explicit `return()` from functions).

Let's mark this as TODO & wait until we get to `explicit_return_linter()`. The logic to identify the last expression in XPath is a bit involved when the last expression is branched. Better to bootstrap off of that.

Reply:
Sounds good. Created a new issue to track this: https://github.com/r-lib/lintr/issues/1574